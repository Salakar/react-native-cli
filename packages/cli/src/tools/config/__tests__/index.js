import path from 'path';
import os from 'os';

import loadConfig from '../';

import {cleanup, writeFiles} from '../../../../../../e2e/helpers';

const DIR = path.resolve(os.tmpdir(), 'resolve_config_path_test');

beforeEach(() => cleanup(DIR));
afterEach(() => cleanup(DIR));

describe('config', () => {
  it('should have a valid structure by default', () => {
    writeFiles(DIR, {
      'package.json': JSON.stringify({
        dependencies: {},
        devDependencies: {},
        'react-native': {reactNativePath: '.'},
      }),
    });
    const {root, ...config} = loadConfig(DIR);
    expect(root).toBe(DIR);
    expect(config).toMatchSnapshot();
  });
});
