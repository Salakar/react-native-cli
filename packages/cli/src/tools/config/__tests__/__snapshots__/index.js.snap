// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should have a valid structure by default 1`] = `
Object {
  "commands": Array [],
  "dependencies": Object {},
  "haste": Object {
    "platforms": Array [],
    "providesModuleNodeModules": Array [],
  },
  "platforms": Object {
    "android": Object {
      "dependencyConfig": [Function],
      "linkConfig": [Function],
      "projectConfig": [Function],
    },
    "ios": Object {
      "dependencyConfig": [Function],
      "linkConfig": [Function],
      "projectConfig": [Function],
    },
  },
  "reactNativePath": ".",
}
`;
